### PR TITLE
Feature: Tick frequency is preserved in .circ file

### PR DESCRIPTION
**Pull request by BFH-ktt1**
(addresses #990)
Step1: the selected tick frequency is saved in the clock component in the .circ file
Step2: restore the saved frequency

DEPENDS_ON #983

### CODE REVIEW

**Comment by BFH-ktt1** [comment]
@maehne, @MarcinOrlowski : I would opt to merge this feature still with 3.6.0, as it is very useful, but I also understand when you do not agree.

**Review by maehne — COMMENTED** [review]
Modifications to the source files look sensible to me. However, I think that they only partly address issue #990 and especially [my comment](https://github.com/logisim-evolution/logisim-evolution/issues/990#issuecomment-912045988) there.

File: src/main/java/com/cburch/logisim/file/XmlWriter.java [hunk 1]
```diff
@@ -243,6 +244,12 @@ void addAttributeSetContent(Element elt, AttributeSet attrs, AttributeDefaultPro
         }
       }
     }
+    if (source instanceof Clock) {
+      final var a = doc.createElement("a");
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/file/XmlWriter.java:248]: No single letter vars. Please rename to i.e. `elementA` or so.

File: src/main/java/com/cburch/logisim/file/XmlReader.java [hunk 2]
```diff
@@ -189,13 +199,13 @@ void initAttributeSet(
             if (IsHolyCross) attrs.setValue(StdAttr.APPEARANCE, StdAttr.APPEAR_CLASSIC);
             else if (IsEvolution) attrs.setValue(StdAttr.APPEARANCE, StdAttr.APPEAR_EVOLUTION);
             else {
-              Object val = defaults.getDefaultAttributeValue(attr, ver);
+              final var val = defaults.getDefaultAttributeValue(attr, ver);
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/file/XmlReader.java:202]: `final var val` that's bit confusing. Mind replacing `val` with i.e. `attrVal`?

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

**Comment by BFH-ktt1** [comment]
removing this PR, as #1018 addresses all features requested.

**Pull request closed without merging**